internal/terraform/exec/exec_mock.go

```diff
@@ -56,7 +56,7 @@ func (mc *MockCall) UnmarshalJSON(b []byte) error {
 		return err
 	}
 
-	mc = (*MockCall)(q.Q[0])
+	*mc = *(*MockCall)(q.Q[0])
```

Comment:
I'm not entirely sure about this one, but just normal assignment here didn't surface this value out of `UnmarshalJSON` so was broken already.

Reply:
If this is broken then I wonder why it doesn't come up in tests 🤔

Reply:
This is because the unmarshaler is never called since we always explicitly unmarshal into `MockItem`:
https://github.com/hashicorp/terraform-ls/blob/592480614510963caf095b7b7e27c6f368b6bfc8/internal/terraform/exec/exec_mock.go#L110-L118

so I think it's better to just remove the unmarshaler entirely.